org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/copyqualifiedname/XbaseCopyQualifiedNameService.java

```diff
@@ -0,0 +1,195 @@
+		} else if (consturctor instanceof JvmOperation && expression instanceof XAbstractFeatureCall) {
+			return _resolveExecutable((JvmOperation) consturctor, (XAbstractFeatureCall) expression, resolvedTypes);
+		} else {
+			throw new IllegalArgumentException(
+					"Unhandled parameter types: " + Arrays.asList(consturctor, expression, resolvedTypes).toString());
+		}
+	}
+
+	protected IResolvedExecutable _resolveExecutable(JvmConstructor consturctor, XAbstractFeatureCall featureCall,
+			IResolvedTypes resolvedTypes) {
+		XExpression actualReceiver = featureCall.getActualReceiver();
+		LightweightTypeReference actualType = null;
+		if (actualReceiver != null) {
+			actualType = resolvedTypes.getActualType(actualReceiver);
+		}
+
+		ResolvedFeatures resolvedFeatures = null;
+		if (actualType != null) {
+			resolvedFeatures = overrideHelper.getResolvedFeatures(actualType);
+		}
+
+		List<IResolvedConstructor> declaredConstructors = null;
+		if (resolvedFeatures != null) {
+			declaredConstructors = resolvedFeatures.getDeclaredConstructors();
+		}
+
+		Optional<IResolvedConstructor> resolvedConstructor = FluentIterable.from(declaredConstructors)
+				.filter(declaredConstructor -> declaredConstructor.getDeclaration().equals(consturctor)) //
+				.first();
+		return resolvedConstructor.isPresent() ? resolvedConstructor.get() : null;
+	}
+
+	protected IResolvedExecutable _resolveExecutable(JvmConstructor constructor, XConstructorCall constructorCall,
+			IResolvedTypes resolvedTypes) {
+		LightweightTypeReference actualType = resolvedTypes.getActualType(constructorCall);
+		ResolvedFeatures resolvedFeatures = null;
+		if (actualType != null) {
+			resolvedFeatures = overrideHelper.getResolvedFeatures(actualType);
+		}
+
+		List<IResolvedConstructor> declaredConstructors = null;
+		if (resolvedFeatures != null) {
+			declaredConstructors = resolvedFeatures.getDeclaredConstructors();
+		}
+
+		Optional<IResolvedConstructor> resolvedConstrutor = FluentIterable.from(declaredConstructors)
+				.filter(constr -> constr.getDeclaration().equals(constructor)) //
+				.first();
+		return resolvedConstrutor.isPresent() ? resolvedConstrutor.get() : null;
+	}
+
+	protected IResolvedExecutable _resolveExecutable(JvmOperation operation, XAbstractFeatureCall featureCall,
+			IResolvedTypes resolvedTypes) {
+		XExpression actualReceiver = featureCall.getActualReceiver();
+		LightweightTypeReference actualType = null;
+		if (actualReceiver != null) {
+			actualType = resolvedTypes.getActualType(actualReceiver);
+		}
+
+		ResolvedFeatures resolvedFeatures = null;
+		if (actualType != null) {
+			resolvedFeatures = overrideHelper.getResolvedFeatures(actualType);
+		}
+
+		List<IResolvedOperation> allOperations = null;
+		if (resolvedFeatures != null) {
+			allOperations = resolvedFeatures.getAllOperations();
+		}
+
+		Optional<IResolvedOperation> resolvedOperation = FluentIterable.from(allOperations)
```

Comment:
Not sure how often we do similar things in the code base, but I can imagine this logic to be moved to `ResolvedFeatures`, e.g. `findResolvedOperation(JvmOperation declaration)`

Reply:
The methods `ResolvedFeatures#getAllOperations()` and `ResolvedFeatures#getDeclaredConstructors()` are used in very few places, not for the same purpose as we have here. Nevertheless, I like you idea of adding methods `#findResolvedOperation(JvmOperation declaration)` (and  `#findDeclaredConstructor(JvmConstructor constructor)`) to the `ResolvedFeatures` class.